### PR TITLE
Skip undefined properties

### PR DESCRIPTION
**Pull request by thomas-ama**
PR for issue #308

### CODE REVIEW

**Comment by thomas-ama** [comment]
PR for issue #308

**Comment by thomas-ama** [comment]
I don't understand why tests are failing on Node.js stable but not on LTS. Any ideas?

**Comment by henrikra** [comment]
This is something we really need also! Have you pinged any maintainers so we get this in?

**Comment by thomas-ama** [comment]
@NoNameProvided @19majkel94

File: src/validation/ValidatorOptions.ts [hunk 1]
```diff
@@ -3,6 +3,11 @@
  */
 export interface ValidatorOptions {
 
+    /**
+     * If set to true than validator will skip validation of all properties that are undefined in the validating object.
+     */
+    skipUndefinedProperties?: boolean;
+
     /**
      * If set to true than validator will skip validation of all properties that are missing in the validating object.
```
Comment (henrikra) [inline src/validation/ValidatorOptions.ts]: Should you also update this documentation about `skipMissingProperties`?

Comment (thomas-ama) [reply src/validation/ValidatorOptions.ts]: What do you mean? `skipMissingProperties` still checks `null` and `undefined` values.
Do you want to replace

> that are missing

with

> that are null or undefined

?
Or do you want to have a new option `skipNullProperties`?

Comment (henrikra) [reply src/validation/ValidatorOptions.ts]: Yes replace missing documentation with `that are null or undefined` or something similar :)

**Comment by henrikra** [comment]
Also if you remove this line https://github.com/typestack/class-validator/blob/master/.travis.yml#L3 then the test will pass becase stable is getting node 12 which is not actually stable :P If you google the error on travis logs you will get lot of cases like this https://www.google.com/search?q=ReferenceError%3A+primordials+is+not+defined&oq=ReferenceError%3A+primordials+is+not+defined&aqs=chrome..69i57j69i58j69i60.220j0j7&sourceid=chrome&ie=UTF-8

File: src/validation/ValidatorOptions.ts (hunk 1 above)
Comment (thomas-ama) [reply src/validation/ValidatorOptions.ts]: Done

**Comment by thomas-ama** [comment]
Thanks @henrikra, let's say tests are okay then.

File: src/validation/ValidatorOptions.ts [hunk 2]
```diff
@@ -4,7 +4,12 @@
 export interface ValidatorOptions {
 
     /**
-     * If set to true than validator will skip validation of all properties that are missing in the validating object.
+     * If set to true than validator will skip validation of all properties that are undefined in the validating object.
```
Comment (henrikra) [inline src/validation/ValidatorOptions.ts]: Both of these lines have typo :D `than` to `then`

**Comment by henrikra** [comment]
This looks golden to me ⭐️

**Comment by AlashaHonore** [comment]
Hello, do you have any idea when this flag will be available ? NEED 😄

**Comment by henrikra** [comment]
Lets merge this!!

**Comment by vlapo** [comment]
@thomas-ama could you please rebase and add also `skipNullProperties` option? We should deprecate `skipMissingProperties` in future and use `skipNullProperties` and `skipUndefinedProperties` instead.

**Comment by ruscon** [comment]
@thomas-ama ping

**Comment by vlapo** [comment]
Close in favore of https://github.com/typestack/class-validator/pull/414. @thomas-ama thank you for your effort.

**Pull request closed without merging**